src/NuGet.Clients/NuGet.PackageManagement.UI/Options/PackageSourcesOptionsControl.cs

```diff
@@ -676,6 +667,42 @@ private static bool IsPathRootedSafe(string path)
             // Otherwise, Path.IsPathRooted() will throw an ArgumentException.
             return path.IndexOfAny(Path.GetInvalidPathChars()) == -1 && Path.IsPathRooted(path);
         }
+
+#nullable enable
+        private async Task<IReadOnlyList<PackageSource>> GetAllPackageSourcesAsync()
```

Comment:
Please add `CancellationToken` parameters to these methods and pass them to all async methods within this one.
Then update callers of this method to provide a CancellationToken (which for now you would get from the JTF.Run overload that you call per another comment I left).